### PR TITLE
IRIS-5039 remove warnings in FounderEmails Extension

### PR DESCRIPTION
**Pull request by sqreek**
https://wikia-inc.atlassian.net/browse/IRIS-5039

Remove warnings from FounderEmailsExtension

@mixth-sense @macbre @TK-999

### CODE REVIEW

File: extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function process( Array $events ) {
 		return true;
 	}
 
-	public static function register( User $user ) {
+	public static function registerUser( User $user = null ) {
```
Comment (macbre) [inline extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php]: Do we want a default value here?

File: extensions/wikia/FounderEmails/FounderEmails.php [hunk 2]
```diff
@@ -53,33 +48,36 @@
...
 	$wgHooks['GetPreferences'][] = 'FounderEmails::onGetPreferences';
 	$wgHooks['UserRights'][] = 'FounderEmails::onUserRightsChange';
 }
 
-$dir = dirname( __FILE__ ) . '/';
+$dir = dirname(__FILE__) . '/';
 $wgAutoloadClasses['FounderEmailsController'] = $dir . 'FounderEmailsController.class.php';
 $wgAutoloadClasses['SpecialFounderEmails'] = $dir . 'SpecialFounderEmails.class.php';
 
+global $wgCityId;
```
Comment (macbre) [inline extensions/wikia/FounderEmails/FounderEmails.php]: `global` not needed - setup files are evaluated in global scope

File: extensions/wikia/FounderEmails/FounderEmails.php [hunk 3]
```diff
@@ -53,33 +48,36 @@
...
-	// add event classes & hooks
-	foreach ( $wgFounderEmailsEvents as $event ) {
-		$wgAutoloadClasses[$event['className']] = $dir . 'events/' . $event['className'] . '.class.php';
-		if ( !empty( $event['hookName'] ) ) {
-			$wgHooks[$event['hookName']][] = $event['className'] . '::register';
-		}
-	}
+	$wgHooks['RecentChange_save'][] = 'FounderEmailsEditEvent::recentChanges';
+	$wgHooks['AddNewAccount'][] = 'FounderEmailsRegisterEvent::registerUser';
+	$wgHooks['CreateWikiLocalJob-complete'][] = 'FounderEmailsDaysPassedEvent::createWikiCompleted';
```
Comment (macbre) [inline extensions/wikia/FounderEmails/FounderEmails.php:72]: :+1:

File: extensions/wikia/FounderEmails/FounderEmails.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/wikia/FounderEmails/FounderEmails.php]: it's still good practice to use explicit scoping rather than rely on implicit global scope

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/FounderEmails/FounderEmails.php [hunk 4]
```diff
@@ -53,33 +48,36 @@
  */
 $wgExtensionFunctions[] = 'wfFounderEmailsInit';
 
-function wfFounderEmailsInit() {
-	global $wgHooks, $wgAutoloadClasses, $wgFounderEmailsEvents, $wgDefaultUserOptions, $wgCityId;
+function wfFounderEmailsInit()
+{
```
Comment (macbre) [inline extensions/wikia/FounderEmails/FounderEmails.php:54]: Do we actually still need this setup function? We do not need to delay the evaluation of anything what's inside this function, right? :)

File: extensions/wikia/FounderEmails/FounderEmails.php (hunk 2 above)
Comment (macbre) [reply extensions/wikia/FounderEmails/FounderEmails.php]: Then we should add `global $wgAutoloadClasses` as well ;)

File: extensions/wikia/FounderEmails/FounderEmails.php (hunk 4 above)
Comment (sqreek) [reply extensions/wikia/FounderEmails/FounderEmails.php:54]: It's `kinda` MW extensions.
If (somehow) it's managed somewhere else (order of execution/logs/disable init) then I will leave it as is

File: extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php (hunk 1 above)
Comment (sqreek) [reply extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php]: fixed in https://github.com/Wikia/app/pull/13954/commits/6a0ec4e6f86d3f04545538e31b59bbfa42e6426a

File: extensions/wikia/FounderEmails/FounderEmails.php [hunk 5]
```diff
@@ -24,27 +26,22 @@
  * extension config
  */
 $wgFounderEmailsEvents = [
-	'edit'       => [
-		'className'  => 'FounderEmailsEditEvent',
-		'hookName'   => 'RecentChange_save',
-		'skipUsers'  => [ 929702 /* CreateWiki script */, 22439 /* Wikia */ ]
+	'edit' => [
+		'className' => 'FounderEmailsEditEvent',
+		'skipUsers' => [929702 /* CreateWiki script */, 22439 /* Wikia */]
```
Comment (macbre) [inline extensions/wikia/FounderEmails/FounderEmails.php:31]: We now have `Fandom` and `FandomBot` users doing automated edits.

Comment (KockaAdmiralac) [reply extensions/wikia/FounderEmails/FounderEmails.php:31]: Did you mean FANDOM and FANDOMbot? Their IDs are [32769624 and 32794352](http://wikia.com/api.php?action=query&list=users&ususers=FANDOM|FANDOMbot&format=jsonfm), if needed.

**Pull request merged**